Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/msg/IConceptMessage.java

```diff
@@ -0,0 +1,65 @@
+package edu.asu.conceptpower.rest.msg;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.validation.ObjectError;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.asu.conceptpower.core.ConceptEntry;
+import edu.asu.conceptpower.core.ConceptType;
+
+/**
+ * This class acts as an interface for all concept message. Currently concepts
+ * are returned in json and xml format. So we have XmlConceptMessage and
+ * JsonConceptMessage implementing this class
+ * 
+ * @author karthik
+ *
+ */
+public interface IConceptMessage {
+
+    /**
+     * This method checks for the validation errors and calls
+     * getAllConceptEntriesAndPaginationDetails for processing.
```

Comment:
That this method calls `getAllConceptEntriesAndPaginationDetails` is an implementation detail that the interface can't promise, because it doesn't implement it.